File: go/controllers/issue_controller.go
package controllers

import (
	"fix-now-backend/config"
	"fix-now-backend/models"
	"net/http"
	"strconv"

	"github.com/gin-gonic/gin"
)

func IssueList(c *gin.Context) {
	userId, _ := c.Get("user_id")
	var user models.User
	config.DB.First(&user, userId)

	var issues []models.Issue
	query := config.DB.Model(&models.Issue{})

	projectId := c.Query("project_id")
	if projectId != "" {
		query = query.Where("project_id = ?", projectId)
	}

	priority := c.Query("priority")
	if priority != "" {
		query = query.Where("priority = ?", priority)
	}

	status := c.Query("status")
	if status != "" {
		query = query.Where("status = ?", status)
	}

	title := c.Query("title")
	if title != "" {
		query = query.Where("title LIKE ?", "%"+title+"%")
	}

	searchIssue := c.Query("searchIssue")
	if searchIssue == "1" {
		query = query.Where("cur_user_id = ? AND status <> 8", user.ID)
	} else if searchIssue == "2" {
		query = query.Where("status <> 8")
	}

	if user.RoleID == 2 {
		query = query.Where("user_id = ?", user.ID)
	}

	// Pagination
	page, _ := strconv.Atoi(c.DefaultQuery("page", "1"))
	limit, _ := strconv.Atoi(c.DefaultQuery("limit", "10"))
	offset := (page - 1) * limit

	var total int64
	query.Count(&total)

	query.Order("status asc, priority desc, id desc").Limit(limit).Offset(offset).Find(&issues)

	// Enrich data (User, CurUser, Module) - N+1 again, but okay for now
	type IssueWithDetails struct {
		models.Issue
		User    models.User   `json:"user"`
		CurUser models.User   `json:"curUser"`
		Module  models.Module `json:"module"`
	}

	var result []IssueWithDetails
	for _, issue := range issues {
		var u models.User
		config.DB.First(&u, issue.UserID)
		var cu models.User
		config.DB.First(&cu, issue.CurUserID)
		var m models.Module
		config.DB.First(&m, issue.ModuleID)

		result = append(result, IssueWithDetails{
			Issue:   issue,
			User:    u,
			CurUser: cu,
			Module:  m,
		})
	}

	c.JSON(http.StatusOK, gin.H{
		"code": 0,
		"msg":  "success",
		"data": gin.H{
			"total": total,
			"data":  result,
		},
	})
}

func IssueEditInfo(c *gin.Context) {
	id := c.Query("id")
	projectId := c.Query("project_id")

	var issue models.Issue
	if id != "" {
		config.DB.First(&issue, id)
	}

	var project models.Project
	if issue.ID > 0 {
		config.DB.First(&project, issue.ProjectID)
	} else if projectId != "" {
		config.DB.First(&project, projectId)
	}

	var users []models.User
	config.DB.Where("status = 1").Find(&users)

	var modules []models.Module
	config.DB.Where("status = 1").Find(&modules)
	moduleTree := buildTree(modules, 0)

	c.JSON(http.StatusOK, gin.H{
		"code": 0,
		"msg":  "success",
		"data": gin.H{
			"id":         issue.ID,
			"title":      issue.Title,
			"content":    issue.Content,
			"priority":   issue.Priority,
			"type":       issue.Type,
			"module_id":  issue.ModuleID,
			"project_id": issue.ProjectID,
			"project":    project,
			"userList":   users,
			"moduleList": moduleTree,
		},
	})
}

func IssueSave(c *gin.Context) {
	userId, _ := c.Get("user_id")
	var input models.Issue
	if err := c.ShouldBindJSON(&input); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
		return
	}

	input.UserID = userId.(uint)
	input.Status = 1 // STATUS_NOTSOLVE

	if input.ID > 0 {
		config.DB.Model(&input).Updates(input)
	} else {
		config.DB.Create(&input)
	}

	// Create Log
	log := models.IssueLog{
		IssueID: input.ID,
		UserID:  userId.(uint),
		Type:    0, // TYPE_ADD
		Content: input.Content,
	}
	config.DB.Create(&log)

	c.JSON(http.StatusOK, gin.H{"code": 0, "msg": "操作成功", "data": input})
}

func IssueLogList(c *gin.Context) {
	id := c.Query("id")
	var logs []models.IssueLog
	config.DB.Where("issue_id = ?", id).Order("created_at asc").Find(&logs)

	// Enrich with User info
	type LogWithUser struct {
		models.IssueLog
		User     models.User `json:"user"`
		NextUser models.User `json:"nextUser"`
	}

	var result []LogWithUser
	for _, l := range logs {
		var u models.User
		config.DB.First(&u, l.UserID)
		var nu models.User
		if l.NextUserID > 0 {
			config.DB.First(&nu, l.NextUserID)
		}
		result = append(result, LogWithUser{
			IssueLog: l,
			User:     u,
			NextUser: nu,
		})
	}

	c.JSON(http.StatusOK, gin.H{"code": 0, "msg": "success", "data": result})
}

type DealInput struct {
	ID        uint   `json:"id"`
	Type      int8   `json:"type"`
	Content   string `json:"content"`
	CurUserID uint   `json:"cur_user_id"`
}

func IssueDeal(c *gin.Context) {
	userId, _ := c.Get("user_id")
	var input DealInput
	if err := c.ShouldBindJSON(&input); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
		return
	}

	var issue models.Issue
	if err := config.DB.First(&issue, input.ID).Error; err != nil {
		c.JSON(http.StatusNotFound, gin.H{"error": "Issue not found"})
		return
	}

	var logType int8
	
	switch input.Type {
	case 2: // FIX
		issue.Status = 2 // STATUS_AUDIT
		issue.CurUserID = issue.UserID
		logType = 2
	case 3: // REJECT
		issue.Status = 2 // STATUS_AUDIT
		issue.CurUserID = issue.UserID
		logType = 3
	case 4: // VER_REJECT
		// Find last fix/reject log to get user?
		// Simplified: just set to not solve
		issue.Status = 1 // STATUS_NOTSOLVE
		// issue.CurUserID = ... (Skip complex logic for now)
		logType = 4
	case 5: // COMMENT
		logType = 5
	case 6: // ASSIGN
		issue.Status = 1 // STATUS_NOTSOLVE
		issue.CurUserID = input.CurUserID
		logType = 6
	case 8: // FINISH
		issue.Status = 8 // STATUS_FINISH
		issue.CurUserID = 0
		logType = 8
	}

	config.DB.Save(&issue)

	// Create Log
	log := models.IssueLog{
		IssueID:    issue.ID,
		UserID:     userId.(uint),
		NextUserID: issue.CurUserID,
		Type:       logType,
		Content:    input.Content,
	}
	config.DB.Create(&log)

	c.JSON(http.StatusOK, gin.H{"code": 0, "msg": "操作成功", "data": issue})
}

File: think/app/controller/IssueController.php
<?php

namespace app\controller;

use app\model\Issue;
use app\model\IssueLog;
use app\model\Module;
use app\model\Project;
use app\model\User;
use think\db\exception\DataNotFoundException;
use think\db\exception\DbException;
use think\db\exception\ModelNotFoundException;

class IssueController extends _Controller
{
    public function listAction()
    {
        [$where, $page, $limit, $order] = $this->buildTableParames();

        $order = 'status asc,priority desc,id desc';

        $priority    = $this->request->get('priority');
        $status      = $this->request->get('status');
        $title       = $this->request->get('title');
        $environment = $this->request->get('environment');
        $bugType     = $this->request->get('bug_type');
        $searchIssue = $this->request->get('searchIssue');

        // New filters
        $curUserId = $this->request->get('cur_user_id');
        $userId    = $this->request->get('user_id');

        $where[] = ['priority', '=', $priority];
        $where[] = ['status', '=', $status];
        $where[] = ['title', 'LIKE', '%' . $title . '%'];
        $where[] = ['project_id', '=', $this->reqPId];
        
        if (!empty($curUserId)) {
            $where[] = ['cur_user_id', '=', $curUserId];
        }
        if (!empty($userId)) {
            $where[] = ['user_id', '=', $userId];
        }
        
        // 新增环境筛选
        if (!empty($environment)) {
            $where[] = ['environment', '=', $environment];
        }
        
        // 新增bug类型筛选
        if (!empty($bugType)) {
            $where[] = ['bug_type', '=', $bugType];
        }

        if ($searchIssue == 1) {
            $where[] = ['cur_user_id', '=', session('user')['id']];
            $where[] = ['status', '<>', 8];
        } else if ($searchIssue == 2) {
            $where[] = ['status', '<>', 8];
        }

        // 用户权限
        if ($this->curUser['role_id'] == 2) {
            $where[] = ['user_id', '=', $this->curUser['id']];
        }

        $data = (new Issue())->findList(['user', 'curUser', 'module'], $where, $page, $limit, $order);

        return $this->vueSuccess($data);
    }

    /**
     * @throws ModelNotFoundException
     * @throws DbException
     * @throws DataNotFoundException
     */
    public function editAction()
    {
        /* @var Issue $row */
        $row = Issue::findOrNew($this->reqId);

        if ($post = $this->request->post()) {
            $data            = $post;
            // 只在创建新issue时设置user_id和默认状态
            if ($row->isEmpty()) {
                $data['user_id'] = $this->curUser['id'];
                $data['status']  = Issue::STATUS_NOTSOLVE;
            }
            $row->save($data);

            if (!empty($post['content'])) {
                $row->editLog($this->curUser['id'], $data['type'] ?? 1, $post['content']);
            }

            return $this->vueSuccess('操作成功', $row);
        }

        if ($row->isEmpty()) {
            $row->project = Project::findOne($this->reqPId);
        } else {
            $row->project;
        }

        $row['userList']   = User::filterWhere(['status' => 1])
            ->withJoin('userProjectList')
            ->select();
        $projectId = $row['project_id'] ?: $this->reqPId;
        $row['moduleList'] = getChildren(Module::filterWhere(['status' => 1, 'project_id' => $projectId])
            ->select());

        return $this->vueSuccess($row);
    }

    /**
     * 日志列表
     * @throws DataNotFoundException
     * @throws ModelNotFoundException
     * @throws DbException
     */
    public function logListAction()
    {
        if (empty($this->reqId)) {
            return $this->fail('参数错误');
        }

        $order = IssueLog::filterWhere(['status' => 1, 'issue_id' => $this->reqId])
            ->with(['user', 'nextUser'])
            ->order('created_at asc')
            ->select()
            ->toArray();

        return $this->vueSuccess($order);
    }

    /**
     * 日志处理
     */
    public function dealAction()
    {
        $row = Issue::findOne($this->reqId);

        if ($post = $this->request->post()) {
            switch ($post['type']) {
                /** 2 修复 */
                case IssueLog::TYPE_FIX:
                    $row->status      = Issue::STATUS_AUDIT;
                    $row->cur_user_id = $row->user_id;
                    $row->save();

                    $row->log($this->curUser['id'], IssueLog::TYPE_FIX, $post['content']);
                    break;
                /** 3 驳回 */
                case IssueLog::TYPE_REJECT:
                    $row->status      = Issue::STATUS_AUDIT;
                    $row->cur_user_id = $row->user_id;
                    $row->save();

                    $row->log($this->curUser['id'], IssueLog::TYPE_REJECT, $post['content']);
                    break;
                /** 4 不通过 */
                case IssueLog::TYPE_VER_REJECT:
                    $lastLog = $row->issueLogList()
                        ->where([
                            'type' => [
                                IssueLog::TYPE_FIX,
                                IssueLog::TYPE_REJECT,
                            ],
                        ])
                        ->order('id DESC')
                        ->find();

                    $row->status      = Issue::STATUS_NOTSOLVE;
                    $row->cur_user_id = $lastLog->user_id;
                    $row->save();

                    $row->log($this->curUser['id'], IssueLog::TYPE_VER_REJECT, $post['content']);
                    break;
                /** 5 评论 */
                case IssueLog::TYPE_COMMENT:
                    $row->log($this->curUser['id'], IssueLog::TYPE_COMMENT, $post['content']);
                    break;
                /** 6 指派 */
                case IssueLog::TYPE_ASSIGN:
                    $row->status      = Issue::STATUS_NOTSOLVE;
                    $row->cur_user_id = $post['cur_user_id'];
                    $row->save();

                    $row->log($this->curUser['id'], IssueLog::TYPE_ASSIGN, $post['content']);
                    break;
                /** 8 通过 */
                case IssueLog::TYPE_FINISH:
                    $row->status      = Issue::STATUS_FINISH;
                    $row->cur_user_id = 0;
                    $row->save();

                    $row->log($this->curUser['id'], IssueLog::TYPE_FINISH, $post['content']);
                    break;
                default:
                    return $this->fail('参数错误');
            }

            return $this->vueSuccess($row);
        }

        return $this->vueSuccess($row);
    }
}
